src/Init/Meta.lean

```diff
@@ -901,22 +901,22 @@ end Syntax
 namespace TSyntax
 
 def getNat (s : NumLit) : Nat :=
-  s.raw.isNatLit?.get!
+  s.raw.isNatLit?.getD 0
 
 def getId (s : Ident) : Name :=
   s.raw.getId
 
 def getScientific (s : ScientificLit) : Nat × Bool × Nat :=
-  s.raw.isScientificLit?.get!
+  s.raw.isScientificLit?.getD (0, false, 0)
```

Comment:
can we have an `Option.getD default` that doesn't panic? This looks a little silly.

Reply:
In this case it actually makes some sense to specify the default value (since the tuple has a special interpretation).  I hope that I picked the equivalent of `0e0` here.

Ideally, we'd have a custom type here though:
```lean
structure ScientificLitDecoded where
  mantissa : Nat
  isExponentNegative : Bool
  exponent : Nat

instance : Inhabited ScientificLitDecoded where
  default := { mantissa := 0, exponent := 0, isExponentNegative := false }
```

Reply:
@gebner I am happy to merge as-is. Do you want to make the change above before merging?